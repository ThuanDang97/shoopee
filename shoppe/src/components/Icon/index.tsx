import React, { memo } from 'react'
import Image, { StaticImageData } from 'next/image'
import { Box, BoxProps } from '@chakra-ui/react'

interface IconProps extends BoxProps {
  srcIcon: StaticImageData | string
  alt: string
}

const Icon = ({ srcIcon, alt, ...rest }: IconProps) => {
  return (
    <Box
      as="figure"
      position="relative"
      display="flex"
      alignItems="flex-end"
      {...rest}
    >
      <Image
        src={srcIcon}
        alt={alt}
        data-testid="image"
        fill
        sizes="(max-width: 768px) 100vw,
              (max-width: 1200px) 50vw,
              33vw"
      />
    </Box>
  )
}

export default memo(Icon)
