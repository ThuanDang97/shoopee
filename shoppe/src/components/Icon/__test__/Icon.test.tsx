import React from 'react'

// component
import Icon from '@components/Icon'

// Utils
import { render } from '@utils/testUtils'

describe('Icon render', () => {
  const props = {
    alt: 'facebook',
    srcIcon: '/images/facebook.webp',
  }

  it('Should show match Icon DOM Snapshot', () => {
    const { container } = render(<Icon {...props} />)
    expect(container).toMatchSnapshot()
  })
})
