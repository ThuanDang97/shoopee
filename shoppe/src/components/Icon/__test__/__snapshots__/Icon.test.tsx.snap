// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render Should show match Icon DOM Snapshot 1`] = `
<div>
  <figure
    class="css-fm4v0q"
  >
    <img
      alt="facebook"
      data-nimg="fill"
      data-testid="image"
      decoding="async"
      loading="lazy"
      sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
      src="/_next/image?url=%2Fimages%2Ffacebook.webp&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Ffacebook.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </figure>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
